CLI tests failing due to typer/click version incompatibility

## Problem
Two CLI tests are failing due to a typer/click version incompatibility:

```
FAILED tests/unit/test_cli.py::TestCLIEdgeCases::test_help_flag_shows_help
FAILED tests/unit/test_cli.py::TestCLIEdgeCases::test_start_help_shows_options

TypeError: Parameter.make_metavar() missing 1 required positional argument: 'ctx'
```

This is a known issue with specific versions of typer and click working together.

## Impact
- Cannot run `--help` on CLI commands
- Tests failing in CI/CD
- User experience degraded

## Suggested Fix
1. Pin typer and click versions in pyproject.toml that are compatible
2. Or update CLI code to work with newer versions
3. Add version constraints to prevent incompatible combinations

```toml
[project.dependencies]
typer = ">=0.9.0,<0.13.0"
click = ">=8.0.0,<9.0.0"
```

## Acceptance Criteria
- [ ] All CLI tests pass
- [ ] `--help` works on all commands
- [ ] Version constraints added to pyproject.toml

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -4,7 +4,8 @@ version = "0.1.0"
 description = "Gastown + OpenClaw + KimiGas in one container"
 requires-python = ">=3.13"
 dependencies = [
-    "typer>=0.21.0",
+    "typer>=0.21.0,<0.25.0",
+    "click>=8.0.0,<9.0.0",
     "httpx>=0.28.0",
     "pydantic>=2.10.0",
     "tomlkit>=0.13.0",

